pytorch_lightning/trainer/data_loading.py

```diff
@@ -88,10 +89,8 @@ def auto_add_sampler(self, dataloader: DataLoader, train: bool) -> DataLoader:
         # don't do anything if it's not a dataloader
         if not isinstance(dataloader, DataLoader):
             return dataloader
-
-        need_dist_sampler = self.use_ddp or self.use_ddp2 or self.use_tpu
-
-        if need_dist_sampler:
+        need_dist_sampler = (self.use_ddp or self.use_ddp2 or self.use_tpu)
+        if self.replace_sampler_ddp and need_dist_sampler:
```

Comment:
Rather than adding a new flag for this, is it possible to just check for the default samplers:
`
(isinstance(dataloader.sampler, RandomSampler) or isinstance(dataloader.sampler, SequentialSampler) or instance(dataloader.sampler, _InfiniteConstantSampler)) and need_dist_sampler?`

I had a hard to find regression that happened because my custom sampler was overridden after #1425. Considering replace_sampler_ddp is set to True by default I think a lot of users will run into similar issues both in terms of regressions and new projects. E.g. users are going to expect that when they write a new sampler and pass it in the dataloader it'll be used without having to change a setting somewhere.

Reply:
I understand that, but I don't think, that this is a good idea. If I explicitly set a standard sampler and don't want it to be replaced, I can't do anything this way.

Reply:
Fair enough. I'd prefer the trainer flag default to False at least, but let's leave it up to the opinion of others @PyTorchLightning/core-contributors